src/sentry/tasks/base.py

```diff
@@ -1,23 +1,30 @@
 from __future__ import annotations
 
+import functools
 import logging
+import random
 from collections.abc import Callable, Iterable
 from datetime import datetime
-from functools import wraps
-from typing import Any, TypeVar
+from typing import Any, ParamSpec, TypeVar
 
-from celery import current_task
+from celery import Task, current_task
 from django.conf import settings
 from django.db.models import Model
 
+from sentry import options
 from sentry.celery import app
 from sentry.silo.base import SiloLimit, SiloMode
+from sentry.taskworker.config import TaskworkerConfig
+from sentry.taskworker.task import Task as TaskWorkerTask
```

Comment:
```suggestion
from sentry.taskworker.task import Task as TaskworkerTask
```

nit: We could make the casing more consistent.